### PR TITLE
BUG: Do not refuse to read images of vectors with many components

### PR DESCRIPTION
**Pull request by dzenanz**
This was encountered when reading 31-component image
which is float on disk, but is being read as double.

https://github.com/dzenanz/ITKUltrasound/blob/398ec5213b3b7d01a769ca6f32b2b36959a30761/test/itkSpectra1DAveragingImageFilterTest.cxx#L155

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)

### CODE REVIEW

**Comment by blowekamp** [comment]
Approach looks good...

File: Modules/IO/ImageBase/include/itkConvertPixelBuffer.hxx [hunk 1]
```diff
@@ -481,6 +490,33 @@ ConvertPixelBuffer<InputPixelType, OutputPixelType, OutputConvertTraits>::Conver
...
+    {
+      OutputConvertTraits::SetNthComponent(c, *outputData, static_cast<OutputComponentType>(*(inputData + c)));
+    }
+    for (int c = componentCount; c < outputNumberOfComponents; ++c)
+    {
+      OutputConvertTraits::SetNthComponent(c, *outputData, 0); // set the rest of components to zero
+    }
+
+    ++outputData;
+    inputData += inputNumberOfComponents;
```
Comment (tbirdso) [inline Modules/IO/ImageBase/include/itkConvertPixelBuffer.hxx:516]: It looks like these two lines are both advancing to the next respective input/output pixel. Just to confirm, is there a reason for the difference in notation?

**Review by tbirdso — COMMENTED** [review]
Just one clarification, otherwise LGTM!

File: Modules/IO/ImageBase/include/itkConvertPixelBuffer.hxx (hunk 1 above)
Comment (dzenanz) [reply Modules/IO/ImageBase/include/itkConvertPixelBuffer.hxx:516]: I was following the pattern established by other similar code in that file.

**Comment by dzenanz** [comment]
I can add the test. The "offending" image is 1x1 pixel with 31 components, so quite small.

**Review by thewtex — APPROVED** [review]

**Comment by thewtex** [comment]
:tada: :taco: :t-rex:

**Pull request merged**